aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -366,16 +391,29 @@ module aptos_framework::fungible_asset {
             balance: 0,
             frozen: false,
         });
+
+        if (default_to_concurrent_fungible_balance()) {
+            move_to(store_obj, ConcurrentFungibleBalance {
+                balance: aggregator_v2::create_unbounded_aggregator(),
+            });
+        };
+
         object::object_from_constructor_ref<FungibleStore>(constructor_ref)
     }
 
     /// Used to delete a store.  Requires the store to be completely empty prior to removing it
-    public fun remove_store(delete_ref: &DeleteRef) acquires FungibleStore, FungibleAssetEvents {
+    public fun remove_store(delete_ref: &DeleteRef) acquires FungibleStore, FungibleAssetEvents, ConcurrentFungibleBalance {
         let store = &object::object_from_delete_ref<FungibleStore>(delete_ref);
         let addr = object::object_address(store);
         let FungibleStore { metadata: _, balance, frozen: _ }
             = move_from<FungibleStore>(addr);
         assert!(balance == 0, error::permission_denied(EBALANCE_IS_NOT_ZERO));
+
+        if (concurrent_fungible_balance_exists(addr)) {
```

Comment:
I would prefer that we have `concurrent_fungible_balance_exists` before you do `assert!(balance == 0, error::permission_denied(EBALANCE_IS_NOT_ZERO));` after unpacking fungible store, otherwise we depend on the fact that the unused `balance` field is always zero? Something like `if .. else ..` reads better?

Reply:
this is checking that both balances are 0, because we are only allowed to remove empty store, so current logic is correct?

Reply:
It is, this is why it is just a my preference (readability wise) but no strong opinion